### ISSUE TITLE
Extra tiles at the edges

### ISSUE THREAD

**Issue by devnoname120** [bvschaik/julius#129]
**Describe the bug**

This is an original bug, it also happens on the unmodified Caesar III.
I have a savegame where extra tiles appear at the edges.

**To Reproduce**
1. Copy the savegame from [Lux-96.zip](https://github.com/bvschaik/julius/files/2984042/Lux-96.zip) in your julius repository.
2. Open it.
3. Scroll to the end of the map.
4. There is a tile outside the map borders, I can't delete it or construct on it.

**Expected behavior**
Only black around the straight map edges.

**Screenshots and saved games**

![julius extra tile issue](https://user-images.githubusercontent.com/2824100/54617318-52073d00-4a61-11e9-9fdd-7e87150fd4ca.gif)

**Additional context**
- Version: https://github.com/bvschaik/julius/commit/5ae213f15a77c879b7e7939224167678016c1520 from AppVeyor
- OS: Windows 10

**Comment by crudelios** [comment]
Could it be because there is no defined entry and exit point? The bugged tiles seem to be close to those points...

**Comment by devnoname120** [comment]
That's possible.

@bvschaik Could you label this issue "Original bug"? I think it will help to distinguish bugs introduced by Julius from original bugs at a glance.

**Comment by bvschaik** [comment]
The cause of these artifacts: terrain is stored as 16-bit mask of terrain flags. For tiles beyond the edge of the map, the value 5 is used, which means tree (1) + water (4). Since these cannot normally occur together, a value of 5 is considered beyond the map limits and is drawn as black tile.
The two tiles instead have value 7: tree (1) + water (4) + rock (2). Which means that the game is drawing nothing at that spot, since it's not drawing a black tile, but also not drawing a normal tile, so the artifacts you see are actually from the previous frame, before you scrolled more black area into view.

Basically it's a small corruption in the file, and it may have already been present in the .scn file. With which game was the city built originally? Do you still have the .scn?

**Comment by devnoname120** [comment]
No, unfortunately I've found this savegame on the Internet when I was looking for big maps to test the framerate of the Vita port.

**Comment by bvschaik** [comment]
Since this is a file corruption rather than an issue with the game itself, I'm going to close this.
